design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,296 @@
+type Customer {
+    gitft_registry_list: [GiftRegistry]
+    gitft_registry(id: ID!): GiftRegistry
+}
+
+type Query {
+    giftRegistryTypes: [GiftRegistryType] @doc(description: "Get a list of available gift registry types")
+    giftRegistrySearch(
+        registrant_firstname: String!,
```

Comment:
Is this secure to allow searching for the gift registry by name or last name?

Reply:
This flow is currently supported by Magento. We also agreed with Nishant to not support search by email and gift registry ID for now.

@nrkapoor do you have any concerns in regards to security?

Reply:
@paliarush - we should not allow search by registry ID and email id. It does not make sense from a storefront standpoint.

Reply:
@nrkapoor what will be the storefront scenario we would like to support? Search by name cannot guarantee us the correctness. Can you please help with the storefront scenario we would like to support?

Reply:
@akaplya there is no use case for storefront for searching by ID or email. Name search may not be accurate but that is the best practice.

Reply:
@akaplya name search will give you a list of records found. It is also possible to adjust search by providing additional search criteria, like date, location etc. The list of additional searchable attribute is configured by the admin.